### ISSUE TITLE
run.dll debugging leftover?

### ISSUE THREAD

**Issue by zx2c4** [lxn/walk#461]
Commit d855149da0e0c79a9bd2fe3add4ac406efbca952, "Add support for widgets that trade height for width to layouts and do some refactoring", made a few changes to sizing logic. But it also strangely added the dynamic loading of "run.dll" and added two new alternative paths, one in the main message loop -- ProcessMessage, replacing the usual Get/Translate/Dispatch loop -- and one regarding applyLayoutResultsRun. What is run.dll? Some sort of debugging leftover of yours? Something else?

**Comment by lxn** [comment]
Good catch!

As indicated by https://github.com/dyu/ffi-overhead, Go FFI has a rather significant overhead. The run.dll was an attempt to replace many win32 syscalls from Go with much fewer syscalls to a small DLL written in Rust, which in turn would do the win32 calls. This improved performance with complex GUIs noticably. For one of my projects, I needed this in the short term, but I wasn't sure if I should release the Rust part.

Some time later I noticed that the Rust implementation had an unfair advantage over the code in walk, and when I removed that disadvantage, the performance of walk seemed to have catched up. This means the experiment will go away soon. If you like, have a look at the attached Rust code.

[run.zip](https://github.com/lxn/walk/files/2900384/run.zip)
